File: guides/java/kora-java-guide-cache-multi-level-app/src/main/java/ru/tinkoff/kora/guide/cache/repository/UserRepository.java
package ru.tinkoff.kora.guide.cache.repository;

import java.util.List;
import java.util.Optional;
import ru.tinkoff.kora.guide.cache.dto.UserResponse;

public interface UserRepository {

    List<UserResponse> findAll();

    Optional<UserResponse> findById(String id);

    String save(String name, String email);

    boolean update(String id, String name, String email);

    boolean deleteById(String id);
}

File: guides/kotlin/kora-kotlin-guide-cache-multi-level-app/src/main/kotlin/ru/tinkoff/kora/guide/cache/repository/UserRepository.kt
package ru.tinkoff.kora.guide.cache.repository

import ru.tinkoff.kora.guide.cache.dto.UserResponse

interface UserRepository {
    fun save(name: String, email: String): String
    fun findById(id: String): UserResponse?
    fun findAll(): List<UserResponse>
    fun update(id: String, name: String, email: String): Boolean
    fun deleteById(id: String): Boolean
}
